include/session/config/convo_info_volatile.hpp

```diff
@@ -0,0 +1,414 @@
+#pragma once
+
+#include <chrono>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <session/config.hpp>
+
+#include "base.hpp"
+
+using namespace std::literals;
+
+extern "C" {
+struct convo_info_volatile_1to1;
+struct convo_info_volatile_open;
+struct convo_info_volatile_legacy_closed;
+}
+
+namespace session::config {
+
+class ConvoInfoVolatile;
+
+/// keys used in this config, either currently or in the past (so that we don't reuse):
+///
+/// Note that this is a high-frequency object, intended only for properties that change frequently (
+/// (currently just the read timestamp for each conversation).
+///
+/// 1 - dict of one-to-one conversations.  Each key is the Session ID of the contact (in hex).
+///     Values are dicts with keys:
+///     r - the unix timestamp (in integer milliseconds) of the last-read message.  Always
+///         included, but will be 0 if no messages are read.
+///     u - will be present and set to 1 if this conversation is specifically marked unread.
+///
+/// o - open group conversations.  This is a nested dict where the outer keys are the BASE_URL of
+///     the open group and the outer value is a dict containing:
+///     - `#` -- the 32-byte server pubkey
```

Comment:
Would it make sense to swap the BASE_URL and the pubkey to prevent duplicates since a sogs can be reached via multiple BASE_URLs ?

Reply:
I'm assuming changing the pubkey of a sogs would break things

Reply:
>I'm assuming changing the pubkey of a sogs would break things
yes, many things would. the stealth addresses ( the `15` prefixed ones ) of all the users would be the most notable of those changes.

Reply:
> Would it make sense to swap the BASE_URL and the pubkey to prevent duplicates since a sogs can be reached via multiple BASE_URLs ?

No: You can use the same pubkey on multiple SOGS instances—and you might want to if, for instance, you decided to split up a single SOGS to multiple SOGS.  But it is impossible to use different pubkeys at the same URL.